modules/relevadRtdProvider.js

```diff
@@ -0,0 +1,367 @@
+            deepSetValue(bid, 'params.target', [].concat(bid.params?.target ? [bid.params.target] : []).concat(relevadList.map(entry => { return 'relevad_rtd=' + entry; })).join(';'));
+            let firstPartyData = {};
+            firstPartyData[bid.bidder] = { firstPartyData: { relevad_rtd: relevadList } };
+            config.setConfig(firstPartyData);
+            !isEmpty(relevadData.segments) && deepSetValue(bid, 'ortb2.user.ext.data.segments', relevadData.segments);
+            !isEmpty(relevadData.categories) && deepSetValue(bid, 'ortb2.user.ext.data.contextual_categories', relevadData.categories.pagecat);
+            !isEmpty(relevadData.categories) && deepSetValue(bid, 'ortb2.site.ext.data.relevad_rtd', relevadData.categories.pagecat);
+            !isEmpty(relevadData.segments) && deepSetValue(bid, 'ortb2.user.ext.data.relevad_rtd', relevadData.segments);
+          }
+        }
+      } catch (e) {
+        logError(e);
+      }
+    });
+  });
+
+  serverData = {...serverData, ...relevadData};
+  return adUnits;
+}
+
+/**
+ * Sends bid info to the RTD server
+ *
+ * @param      {JSON}  data  Bids information
+ * @param      {object}  config  Configuraion
+ */
+function sendBids(data, config) {
+  let dataJson = JSON.stringify(data);
+
+  if (!config.dryrun) {
+    ajax(RELEVAD_API_DOMAIN + '/apis/bids/', () => {}, dataJson, AJAX_OPTIONS);
+  }
+};
+
+/**
+ * Processes AUCTION_END event
+ *
+ * @param      {object}  auctionDetails  Auction details
+ * @param      {object}  config          Module configuration
+ * @param      {object}  userConsent     User GDPR consent object
+ */
+function onAuctionEnd(auctionDetails, config, userConsent) {
+  let adunitObj = {};
+  let adunits = [];
+
+  // Add Bids Received
+  auctionDetails.bidsReceived.forEach((bidObj) => {
+    if (!adunitObj[bidObj.adUnitCode]) { adunitObj[bidObj.adUnitCode] = []; }
+
+    adunitObj[bidObj.adUnitCode].push({
+      bidder: bidObj.bidderCode || bidObj.bidder,
+      cpm: bidObj.cpm,
+      currency: bidObj.currency,
+      dealId: bidObj.dealId,
+      type: bidObj.mediaType,
+      ttr: bidObj.timeToRespond,
+      size: bidObj.size
+    });
+  });
+
+  entries(adunitObj).forEach(([adunitCode, bidsReceived]) => {
+    adunits.push({code: adunitCode, bids: bidsReceived});
+  });
+
+  let data = {
+    event: 'bids',
+    adunits: adunits,
+    reledata: serverData.rawdata,
+    pid: encodeURIComponent(config.params?.publisherid || ''),
+    aid: encodeURIComponent(config.params?.apikey || ''),
+    cid: encodeURIComponent(config.params?.partnerid || ''),
+    gdpra: encodeURIComponent(userConsent?.gdpr?.gdprApplies || ''),
+    gdprc: encodeURIComponent(userConsent?.gdpr?.consentString || ''),
+  }
+  if (!config.dryrun) {
+    data.page = serverData?.page || config?.params?.actualUrl || getRefererInfo().page || '';
+  }
+
+  sendBids(data, config);
+
+  serverData = {};
+  return data;
```

Comment:
I don't believe `onAuctionEndEvent` expects a return value

Reply:
Fixed, thank you!